ipn/localapi/localapi.go

```diff
@@ -1037,6 +1038,69 @@ func (h *Handler) serveTKAModify(w http.ResponseWriter, r *http.Request) {
 	w.Write(j)
 }
 
+// serveProfiles serves profile switching-related endpoints. Supported methods
+// and paths are:
+// - GET /profiles/: list all profiles (JSON-encoded array of ipn.LoginProfiles)
+// - PUT /profiles/: add new profile (no response)
```

Comment:
this is a weird one. No request body and no response about what it auto-generated for you? Does it get a new random ID, or?

Reply:
It is a bit weird because of the semantics where the profile is not persisted till the user logs in.
If the user switches out of the profile without logging in, they won't be able to switch back to the "empty" profile they previously created.

Reply:
Added a comment to say that `StartLoginInteractive()` or similar is needed to actually populate and persist the profile.

Reply:
Added an error check and changed the response type to `StatusCreated`.